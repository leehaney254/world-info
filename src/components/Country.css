.flagSize {
  width: 7rem;
  height: 5rem;
}

.homeFlag {
  display: flex;
  justify-content: space-around;
}

.countryName {
  display: flex;
  flex-direction: column;
  align-items: flex-end;
}

.countryPopulation {
  display: flex;
}

.countryLink {
  text-decoration: none;
  color: white;
}

.borders {
  border: solid 0.1px white;
  padding: 0.5rem;
}

.colon {
  margin: 0 0.2rem;
}
