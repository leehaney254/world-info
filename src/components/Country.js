import React from 'react';
import PropTypes from 'prop-types';
import { BsArrowRightCircle } from 'react-icons/bs';
import { Link } from 'react-router-dom';

const Country = ({ name, population }) => (
  <div>
    <Link to={name}>
      <div>
        Flag
        <BsArrowRightCircle />
      </div>
      <p>{name}</p>
      <p>{population}</p>
    </Link>
  </div>
);

Country.propTypes = {
  name: PropTypes.string.isRequired,
  population: PropTypes.number.isRequired,
};

export default Country;
