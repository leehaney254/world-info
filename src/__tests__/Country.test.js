import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from '@cfaester/enzyme-adapter-react-18';
import Country from '../components/Country';

configure({ adapter: new Adapter() });

describe('Country component', () => {
  it('renders correctly', () => {
    const wrapper = shallow(
      <Country
        name="Kenya"
        population="53771300"
        flag="https://flagcdn.com/ke.svg"
      />,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('renders correctly', () => {
    const wrapper = shallow(
      <Country
        name="Uganda"
        population="5399000"
        flag="https://flagcdn.com/ke.svg"
      />,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
