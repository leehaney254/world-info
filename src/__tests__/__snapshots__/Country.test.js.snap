// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Country component renders correctly 1`] = `ShallowWrapper {}`;

exports[`Country component renders correctly 2`] = `ShallowWrapper {}`;

exports[`Country render component The render is correct 1`] = `
<div>
  <a
    href="/Kenya"
    onClick={[Function]}
  >
    <div>
      Flag
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 16 16"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M1 8a7 7 0 1 0 14 0A7 7 0 0 0 1 8zm15 0A8 8 0 1 1 0 8a8 8 0 0 1 16 0zM4.5 7.5a.5.5 0 0 0 0 1h5.793l-2.147 2.146a.5.5 0 0 0 .708.708l3-3a.5.5 0 0 0 0-.708l-3-3a.5.5 0 1 0-.708.708L10.293 7.5H4.5z"
          fillRule="evenodd"
        />
      </svg>
    </div>
    <p>
      Kenya
    </p>
    <p>
      50000000
    </p>
  </a>
</div>
`;

exports[`Country render component The render is correct 2`] = `
<div>
  <a
    href="/Uganda"
    onClick={[Function]}
  >
    <div>
      Flag
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 16 16"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M1 8a7 7 0 1 0 14 0A7 7 0 0 0 1 8zm15 0A8 8 0 1 1 0 8a8 8 0 0 1 16 0zM4.5 7.5a.5.5 0 0 0 0 1h5.793l-2.147 2.146a.5.5 0 0 0 .708.708l3-3a.5.5 0 0 0 0-.708l-3-3a.5.5 0 1 0-.708.708L10.293 7.5H4.5z"
          fillRule="evenodd"
        />
      </svg>
    </div>
    <p>
      Uganda
    </p>
    <p>
      35000000
    </p>
  </a>
</div>
`;
